include/albatross/src/cereal/variant.hpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2019 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef INCLUDE_ALBATROSS_SRC_CEREAL_VARIANT_HPP_
+#define INCLUDE_ALBATROSS_SRC_CEREAL_VARIANT_HPP_
+
+/*
+ * A lot of this is borrowed from cereal's std::variant.
+ */
+
+namespace cereal {
+
+namespace mapbox_variant_detail
+{
+
+  template <typename Variant>
+  struct variant_size {};
+
+  template <typename... VariantTypes>
+  struct variant_size<variant<VariantTypes...>> : public std::tuple_size<std::tuple<VariantTypes...>> {};
+
+  template<int N, class Variant, class ... Args, class Archive,
+           typename std::enable_if_t<N == variant_size<Variant>::value, int> = 0>
+  void load_variant(Archive &, int, Variant &) {
+    assert(false); // load_variant received an out of bounds index.
```

Comment:
Did you try adding a static assert here? I'm not sure if it would just always fail.

Reply:
yeah, that'd always fail since this is always included as the end of the recurssion.